Compilation issue in Prophet

The build in main currently fails with the error `TypeError: object.__init__() takes exactly one argument (the instance to initialize)`

New holidays error based on the way we import prophet:  

```
_____________________________________________________________________________________________________ ERROR collecting tests/test_prophet.py _____________________________________________________________________________________________________
test_prophet.py:8: in <module>
    from hummingbird.ml._utils import pandas_installed, prophet_installed, onnx_runtime_installed
../hummingbird/ml/__init__.py:15: in <module>
    from . import supported as hummingbird_constants
../hummingbird/ml/supported.py:513: in <module>
    prophet_operator_list = _build_prophet_operator_list()
../hummingbird/ml/supported.py:377: in _build_prophet_operator_list
    if prophet_installed():
../hummingbird/ml/_utils.py:191: in prophet_installed
    from prophet import Prophet
../../.local/lib/python3.8/site-packages/prophet/__init__.py:8: in <module>
    from prophet.forecaster import Prophet
../../.local/lib/python3.8/site-packages/prophet/forecaster.py:17: in <module>
    from prophet.make_holidays import get_holiday_names, make_holidays_df
../../.local/lib/python3.8/site-packages/prophet/make_holidays.py:14: in <module>
    import prophet.hdays as hdays_part2
../../.local/lib/python3.8/site-packages/prophet/hdays.py:683: in <module>
    class TU(Turkey):
../../.local/lib/python3.8/site-packages/holidays/registry.py:177: in __init__
    raise TypeError(
E   TypeError: This is a python-holidays entity loader class. For entity inheritance purposes please import a class you want to derive from directly: e.g., `from holidays.countries import Entity` or `from holidays.financial import Entity`.
============================================================================================================ short test summary info =============================================================================================================
ERROR test_prophet.py - TypeError: This is a python-holidays entity loader class. For entity inheritance purposes please import a class you want to derive from directly: e.g., `from holidays.countries import Entity` or `from holidays.finan...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================================================ 1 error in 2.74s ============================================================
```

--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,7 @@ extra_requires = [
     # The need each for these depends on which libraries you plan to convert from
     "xgboost>=0.90,<2.0.0",
     "lightgbm>=2.2",
-    "holidays==0.24",
+    "holidays",
     "prophet",
 ]
 setup(

